api/internal/localizer/locloader.go

```diff
@@ -0,0 +1,382 @@
+	if err != nil {
+		log.Fatalf(errors.WrapPrefixf(
+			err,
+			"%s: cannot parse validated remote root '%s'", fatalPrefix, rootURL).Error())
+	}
+	if spec.Ref == "" {
+		return nil, errors.WrapPrefixf(
+			errors.Errorf("localize remote root missing ref query string parameter"),
+			"invalid localize remote root '%s'", rootURL)
+	}
+	return spec, nil
+}
+
+// urlBase is the url equivalent of filepath.Base
+func urlBase(url string) string {
+	cleaned := strings.TrimRight(url, "/")
+	i := strings.LastIndex(cleaned, "/")
+	if i < 0 {
+		return cleaned
+	}
+	return cleaned[i+1:]
+}
+
+// defaultNewDir calculates the default localize destination directory name from targetLdr at the localize target
+// and spec of target, which is nil if target is local
+func defaultNewDir(targetLdr ifc.Loader, spec *git.RepoSpec) string {
+	targetDir := filepath.Base(targetLdr.Root())
+	if repo, isRemote := targetLdr.Repo(); isRemote {
+		// kustomize doesn't download repo into repo-named folder
+		// must find repo folder name from url
+		if repo == targetLdr.Root() {
+			targetDir = urlBase(spec.OrgRepo)
+		}
+		return strings.Join(
+			[]string{dstPrefix, targetDir, strings.ReplaceAll(spec.Ref, "/", "-")},
+			"-")
+	}
+	// special case for local target directory since destination directory cannot have "/" in name
+	if targetDir == string(filepath.Separator) {
+		return dstPrefix
+	}
+	return strings.Join([]string{dstPrefix, targetDir}, "-")
+}
+
+// validateNewDir returns the localize destination directory or error. Note that spec is nil if targetLdr is at local
+// target.
+func validateNewDir(newDirArg string, targetLdr ifc.Loader, spec *git.RepoSpec, fSys filesys.FileSystem) (filesys.ConfirmedDir, error) {
+	var newDirPath string
+	if newDirArg == "" {
+		newDirPath = defaultNewDir(targetLdr, spec)
+	} else {
+		newDirPath = newDirArg
+	}
+	if fSys.Exists(newDirPath) {
+		return "", errors.WrapPrefixf(
+			errors.Errorf("localize destination already exists"),
+			"invalid localize destination path '%s'", newDirPath)
+	}
+	// destination directory must sit in existing directory
+	if err := fSys.Mkdir(newDirPath); err != nil {
+		return "", errors.WrapPrefixf(err, "unable to create localize destination directory")
+	}
+	newDir, err := filesys.ConfirmDir(fSys, newDirPath)
+	if err != nil {
+		if err := fSys.RemoveAll(newDir.String()); err != nil {
+			log.Printf("%s", errors.WrapPrefixf(err, errCleanDst).Error())
+		}
+		return "", errors.WrapPrefixf(err, "unable to establish localize destination")
+	}
+
+	return newDir, nil
+}
+
+// pathSuffix returns path relative to prefix, where both are cleaned absolute paths
+// and prefix is a directory known to contain path
+func pathSuffix(prefix string, path string) string {
+	if prefix == string(filepath.Separator) || prefix == path {
+		return path[len(prefix):]
+	}
+	return path[len(prefix)+1:]
+}
+
+// Root returns the path to the current target directory
+func (ll *LocLoader) Root() string {
+	return ll.ldr.Root()
```

Comment:
nit: suggest embedding the `ifc.Loader` so that we don't need this method.

Reply:
Embedding is done by just not naming the variable in the struct. Example:

```
type A struct {
  B
}
```

AFAIK this is the closest thing that Go has to inheritance. A will inherit all of B's methods, i.e. if B has a function called `func`, then `A.func()` is equivalent to `A.B.func()`.